sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -249,57 +249,51 @@ private[sbtplugin] object ScalaJSPluginInternal {
       legacyKey := {
         val linkingResult = key.value
 
-        val module = {
-          val report = linkingResult.data
-
-          if (report.publicModules.size != 1) {
-            throw new MessageOnlyException(
-                "Linking did not return exactly one public module. " +
-                s"${legacyKey.key} can only deal with a single module. " +
-                s"Did you mean to invoke ${key.key} instead? " +
-                s"Full report:\n$report")
-          }
+        val linkerImpl = (scalaJSLinkerImpl in key).value
+
+        val report = linkingResult.data
+        val outDir =
+          linkerImpl.outputDirectory(linkerOutputDirectory(linkingResult).toPath())
+
+        val outputJSFile = (artifactPath in legacyKey).value
+        val outputSourceMapFile = new File(outputJSFile.getPath + ".map")
+
+        IO.createDirectory(outputJSFile.getParentFile)
```

Comment:
Is this line the hidden bug fix?

Reply:
Nope. That's also a bug fix indeed, but its untested :-/